pkg/cmd/cli/secrets/add_secret_to_obj.go

```diff
@@ -181,7 +186,38 @@ func (o AddSecretOptions) AddSecrets() error {
 	return nil
 }
 
-func (o AddSecretOptions) getSecrets() ([]*api.Secret, error) {
+// TODO: when Secrets in kapi.ServiceAccount get changed to MountSecrets and represented by LocalObjectReferences, this can be
+// refactored to reuse the addition code better
+// addSecretsToServiceAccount adds secrets to the service account, either as pull secrets, mount secrets, or both.
+func (o AddSecretOptions) addSecretsToServiceAccount(serviceaccount *kapi.ServiceAccount) error {
+	newSecrets, err := o.getSecrets()
+	if err != nil {
+		return err
+	}
+	newSecretNames := getSecretNames(newSecrets)
+
+	if !o.ForPull && !o.ForMount {
+		return errors.New("this type of secret type is not handled for ServiceAccounts")
+	}
+	if o.ForMount {
+		currentSecrets := getMountSecretNames(serviceaccount)
+		secretsToAdd := newSecretNames.Difference(currentSecrets)
+		for _, secretName := range secretsToAdd.List() {
+			serviceaccount.Secrets = append(serviceaccount.Secrets, kapi.ObjectReference{Name: secretName})
+		}
+	}
+	if o.ForPull {
+		currentSecrets := getPullSecretNames(serviceaccount)
+		secretsToAdd := newSecretNames.Difference(currentSecrets)
+		for _, secretName := range secretsToAdd.List() {
+			serviceaccount.ImagePullSecrets = append(serviceaccount.ImagePullSecrets, kapi.LocalObjectReference{Name: secretName})
+		}
+	}
+	_, err = o.ClientInterface.ServiceAccounts(o.Namespace).Update(serviceaccount)
```

Comment:
only do the update if any changes were actually made to either the pull or mount secrets lists